.github/workflows/test_libs.yml

```diff
@@ -0,0 +1,204 @@
+      tkassert: ${{ steps.filter.outputs.tkassert }}
+      tkrng: ${{ steps.filter.outputs.tkrng }}
+      libs: ${{ steps.filter.outputs.changes }}
+    steps:
+    - uses: actions/checkout@v3
+    - uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        base: ${{ github.ref }}
+        filters: |
+          tklog:
+            - 'libs/tklog/**'
+          tkassert:
+            - 'libs/tkassert/**'
+          tkrng:
+            - 'libs/tkrng/**'
+          tktokenswap:
+            - 'libs/tktokenswap/**'
+  test_libraries:
+    name: test library
+    needs: changes
+    if: ${{ needs.changes.outputs.libs != '[]' && needs.changes.outputs.libs != '' }}
+    strategy:
+      matrix:
+        os: ['ubuntu-20.04', 'macos-11', 'windows-2019']
+        lib: ${{ fromJson(needs.changes.outputs.libs) }}
+    runs-on: ${{ matrix.os }}
+    env:
+      CONAN_REVISIONS_ENABLED: 1
+    steps:
+    - uses: actions/checkout@v3
+    - name: set compiler
+      if: matrix.os == 'ubuntu-20.04'
+      run: |
+        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 100
+        sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-10 100
+        sudo update-alternatives --set gcc /usr/bin/gcc-10
+        sudo update-alternatives --set g++ /usr/bin/g++-10
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v3
+      with:
+        python-version: '3.9'
+    - name: install conan
+      run: pip install conan
+    - name: create profile
+      run: conan profile new tket --detect
+    - name: set libcxx
+      if: matrix.os == 'ubuntu-20.04'
+      run: conan profile update settings.compiler.libcxx=libstdc++11 tket
+    - name: add remote
+      run: conan remote add tket-conan https://tket.jfrog.io/artifactory/api/conan/tket-conan
+    - name: build ${{ matrix.lib }}
+      run: conan create --profile=tket libs/${{ matrix.lib }} --build=missing
+    - name: build ${{ matrix.lib }} tests
+      run: |
+        conan install libs/${{ matrix.lib }}/test --install-folder=build/test-${{ matrix.lib }} --profile=tket
+        conan build libs/${{ matrix.lib }}/test --configure --build-folder=build/test-${{ matrix.lib }} --source-folder=libs/${{ matrix.lib }}/test
+        conan build libs/${{ matrix.lib }}/test --build --build-folder=build/test-${{ matrix.lib }}
+        conan package libs/${{ matrix.lib }}/test --build-folder=build/test-${{ matrix.lib }} --package-folder build/test-${{ matrix.lib }}/package --source-folder=libs/${{ matrix.lib }}/test
+    - name: run ${{ matrix.lib }} tests
+      working-directory: ./build/test-${{ matrix.lib }}/package/bin
+      run: ./test-${{ matrix.lib }}
+  macos-m1:
+    name: test library (macos-m1)
+    needs: changes
+    if: ${{ needs.changes.outputs.libs != '[]' && needs.changes.outputs.libs != '' }}
+    runs-on: [self-hosted, macos, M1]
+    strategy:
+      matrix:
+        lib: ${{ fromJson(needs.changes.outputs.libs) }}
```

Comment:
Yes, only changed libs are tested. There is currently no scheduled build. I'm not sure there is much need since this already tests all configurations, but it's true that occasionally build failures can happen for external reasons, so it might be worth it.

Reply:
I have added a scheduled test of all libraries. Workflow tested [here](https://github.com/CQCL/tket/actions/runs/2542057515).